### PR TITLE
[ENH] Improves `create_multi_comparison_matrix` parameters and saving

### LINKED ISSUE #2650: [ENH] Improve `create_multi_comparison_matrix` parameters and saving
### Describe the feature or idea you want to propose

The MCM figure creator has a lot of parameters, some of which could be a lot tidier IMO. i.e. we probably dont need 5 (or more) parameters for output. 

```
    output_dir="./",
    pdf_savename=None,
    png_savename=None,
    csv_savename=None,
    tex_savename=None,
```

Same with ones like `order_win_tie_loss` and `order_better` which feel like they could be the same.

Saving the returned figure as a PDF also just returns a blank file. Not sure if plotting the returned figure is any better.

```
    mcm = create_multi_comparison_matrix(df)
    mcm.savefig(
        f"{save_path}/{statistic_name}/figures/"
        f"{eval_name}_{statistic_name.lower()}_mcm.pdf",
        bbox_inches="tight",
    )
```

### Describe your proposed solution

Improve saving for the MCM figure and condense parameters, or at least discuss doing so.

### Describe alternatives you've considered, if relevant

_No response_

### Additional context

I was trying to use it in my evaluation package and it was a little unintuitive to get working IMO 🙂. https://github.com/time-series-machine-learning/tsml-eval/blob/main/tsml_eval/evaluation/multiple_estimator_evaluation.py#L1330

### PR DESCRIPTION
**Pull request by aryanpola**
#### Reference Issues/PRs
Fixes #2650 

#### What does this implement/fix? Explain your changes.

Condenses parameters into dictionaries.
1. ```     
    output_dir="./",
    pdf_savename=None,
    png_savename=None,
    csv_savename=None,
    tex_savename=None,
   ```
  into
  ```
  output_config=None,
  ```
2. ```
    order_win_tie_loss="higher",
    order_better="decreasing",
    ```
into 
```
ordering_strategy="higher-wins"
```
3. ```
    win_label="r>c",
    tie_label="r=c",
    loss_label="r<c",
    ```
into
```
result_labels=True
```

Also, this fixes the output. Output will now be shown in the pdf.
![Screenshot 2025-03-20 114404](https://github.com/user-attachments/assets/0984bba6-450e-47dc-bb76-003f719fd884)

### CODE REVIEW

**Comment by aryanpola** [comment]
Please let me know if there are any more output parameters that can be reduced.

**Comment by TonyBagnall** [comment]
thanks for this and sorry for the lack of review. Personally I think I prefer the current version, wrapping in a dictionary seems a bit over complex, given they all default to not saving. @hadifawaz1999 what do you think?

File: aeon/visualisation/results/_mcm.py [hunk 1]
```diff
@@ -63,28 +56,23 @@ def create_multi_comparison_matrix(
...
-        The name of the saved file into csv format, if None, will not be saved into
-        this format.
-    tex_savename: str, default = None
-        The name of the saved file into tex format, if None, will not be saved into
-        this format.
+    output_config: dict, default = None
+        A dictionary for output configurations. Expected keys:
+            - "dir": output directory (default: "./")
+            - "base_name": basef filename (if provided, used for saving)
+            - "format": list of file formats to save. eg:["pdf", "png", "csv", "tex"]
```
Comment (MatthewMiddlehurst) [inline aeon/visualisation/results/_mcm.py:63]: I think keep output_dir, or possibly rename save_path. I dont see the need for four different parameters for formats though. Just have 4 str inputs and output one. It is not expensive enough to run that you need to do all of them at once (and im not sure why you would want it as a csv or tex)

File: aeon/visualisation/results/_mcm.py [hunk 2]
```diff
@@ -63,28 +56,23 @@ def create_multi_comparison_matrix(
...
+    ordering_strategy : str, default="higher-wins"
+        Unified parameter to specify ordering.
+            - "higher-wins": higher metric values (accuracy) count as wins and sorting
+            is in decreasing order.
+            - "lower-wins": lower metric values (error) count as wins and sorting is in
+               increasing order.
```
Comment (MatthewMiddlehurst) [inline aeon/visualisation/results/_mcm.py:69]: This would be better as a bool imo. Please order these the same as the init parameter order.

File: aeon/visualisation/results/_mcm.py [hunk 3]
```diff
@@ -147,12 +133,10 @@ def create_multi_comparison_matrix(
...
-    win_label: str, default = "r>c"
-        The winning label to be set on the MCM.
-    tie_label: str, default = "r=c"
-        The tie label to be set on the MCM.
-    loss_label: str, default = "r<c"
-        The loss label to be set on the MCM.
+    result_labels : bool, default=True
+        r = row estimator performance, c = column estimator performance.
+        If True, use default labels (win: "r>c", tie: "r=c", loss: "r<c");
+        if False, use the reversed labels (win: "r<c", tie: "r=c", loss: "r>c").
```
Comment (MatthewMiddlehurst) [inline aeon/visualisation/results/_mcm.py:139]: maybe also accept list/tuple of lenth 3?

File: aeon/visualisation/results/_mcm.py (hunk 1 above)
Comment (hadifawaz1999) [reply aeon/visualisation/results/_mcm.py:63]: No I would prefer keeping csv and tex, we added it for a reason as not everyone wants the PDF, csv could be used to make their own figure, and tex is to have it directly in tikz format

**Comment by hadifawaz1999** [comment]
> thanks for this and sorry for the lack of review. Personally I think I prefer the current version, wrapping in a dictionary seems a bit over complex, given they all default to not saving. @hadifawaz1999 what do you think?

i agree with @TonyBagnall to be honest

**Review by hadifawaz1999 — CHANGES_REQUESTED** [review]
Am not a fan of replacing the order win tie loss and the order better parameters by the ordering_strategy parameter. The MCM has a goal of giving the user ability to manipulate the setup as they wish, the win tie loss can be on MSE where "lower" should be used and the order better should be "decreasing" if the order_stats parameter is set to max-wins

I prefer not changing the dual-parameter setup for those, happy to change naming and/or string possibilities to avoid confusing but not the functionality as is

**Comment by aryanpola** [comment]
> Hi @aryanpola, had a chat and I think these are the changes we would want for this PR:
> 
> * `output_dir` should be refactored into `save_path` including a filename
> * `pdf_savename`, `png_savename`, `csv_savename`, `tex_savename` and `save_as_json` should be a single str parameter
> * `order_win_tie_loss` should be a bool and renamed i.e. `higher_stat_better`
> * `order_better` again should probably be a boolean and called something like `order_stats_increasing`
> * `win_label`, `tie_label`, `loss_label` should be a single parameter IMO, have it as a list/tuple and check for length and contents. Either way it should default to None and change based on `order_win_tie_loss`

Just need a small clarification on point 2. Would you like to save all the formats, or would the user type in a list of formats (as a parameter)?

```
create_multi_comparison_matrix(results, save_path="outputs/mcm", formats=("png", "json"))
```

File: aeon/visualisation/results/_mcm.py [hunk 4]
```diff
@@ -57,27 +57,18 @@ def create_multi_comparison_matrix(
         A csv file containing results in `n_problems,n_estimators` format. The first
         row should contain the names of the estimators and the first column can
         contain the names of the problems if `dataset_column` is true.
-    save_path: str, default = './'
+    save_path: str, default = './mcm'
         The output directory for the results.
```
Comment (MatthewMiddlehurst) [inline aeon/visualisation/results/_mcm.py:61]: mention that a file name is required at the end of the directory path if that is the case

File: aeon/visualisation/results/_mcm.py [hunk 5]
```diff
@@ -142,12 +133,9 @@ def create_multi_comparison_matrix(
...
         The values for which the heat map colors are based on.
-    win_label: str, default = "r>c"
-        The winning label to be set on the MCM.
-    tie_label: str, default = "r=c"
-        The tie label to be set on the MCM.
-    loss_label: str, default = "r<c"
-        The loss label to be set on the MCM.
+    labels: tuple of str, default = None
+        The labels for the heatmap cells, if labels=None and higher_stat_better=True,
+        then the labels are ('r>c', 'r=c', 'r<c'), otherwise ('r<c', 'r=c', 'r>c').
```
Comment (MatthewMiddlehurst) [inline aeon/visualisation/results/_mcm.py:138]: "tuple of str or None".

This documentation could be expanded. It does not mention tuple input and just "otherwise" is a bit ambiguous.

File: aeon/visualisation/results/_mcm.py [hunk 6]
```diff
@@ -147,12 +133,9 @@ def create_multi_comparison_matrix(
         In which orientation to show the colorbar either horizontal or vertical.
     colorbar_value: str, default = 'mean-difference'
         The values for which the heat map colors are based on.
-    win_label: str, default = "r>c"
-        The winning label to be set on the MCM.
-    tie_label: str, default = "r=c"
-        The tie label to be set on the MCM.
-    loss_label: str, default = "r<c"
-        The loss label to be set on the MCM.
+    labels: tuple of str, default = None
```
Comment (MatthewMiddlehurst) [inline aeon/visualisation/results/_mcm.py:136]: `labels` is a bit ambiguous. Maybe `win_draw_loss_labels`?

**Review by MatthewMiddlehurst — CHANGES_REQUESTED** [review]
This looks fine other than a few comments. I am going to add the coverage tag just to make sure these changes are covered by testing.

File: examples/visualisation/plotting_results.ipynb [hunk 7]
```diff
@@ -16,7 +16,7 @@
   },
```
Comment (MatthewMiddlehurst) [inline examples/visualisation/plotting_results.ipynb:16]: <p>Not sure why the above plot was removed?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aeon-toolkit/aeon/pull/2657/discussion/#comment-2117922245'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='8'/>

Comment (MatthewMiddlehurst) [inline examples/visualisation/plotting_results.ipynb:16]: <p>It seems to be plotting fine in the notebook before?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aeon-toolkit/aeon/pull/2657/discussion/#comment-2117922246'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='17'/>

Comment (aryanpola) [reply examples/visualisation/plotting_results.ipynb:16]: <p>Running the cell produces 2 identical plots. Now, it produces a single plot.</p>

Comment (aryanpola) [reply examples/visualisation/plotting_results.ipynb:16]: Should be sorted now.

**Comment by MatthewMiddlehurst** [comment]
A lot of this seems untested though https://app.codecov.io/gh/aeon-toolkit/aeon/pull/2657/blob/aeon/visualisation/results/_mcm.py

File: aeon/visualisation/results/_mcm.py [hunk 8]
```diff
@@ -963,14 +958,14 @@ def _get_pairwise_content(
...
 
-    if order_WinTieLoss == "lower":
-        win = len(x[x < y])
-        loss = len(x[x > y])
+    if higher_stat_better is True:
+        win = len(x[x > y])
+        loss = len(x[x < y])
         tie = len(x[x == y])
 
     else:  # by default we assume higher is better
```
Comment (hadifawaz1999) [inline aeon/visualisation/results/_mcm.py]: i would remove the comment here in front of else

File: aeon/visualisation/results/_mcm.py [hunk 9]
```diff
@@ -16,23 +16,19 @@
 
 def create_multi_comparison_matrix(
     df_results,
-    output_dir="./",
-    pdf_savename=None,
-    png_savename=None,
-    csv_savename=None,
-    tex_savename=None,
+    save_path="./mcm",
+    formats=("pdf", "png", "csv", "tex", "json"),
```
Comment (hadifawaz1999) [inline aeon/visualisation/results/_mcm.py]: i think its best to have a default not to save anything no ? i recall @TonyBagnall made it like that when porting it into aeon, unless i missunderstood the code and it already does what am saying ! but if not then i would default it to None

Comment (hadifawaz1999) [reply aeon/visualisation/results/_mcm.py]: what do you think @MatthewMiddlehurst ?

**Review by hadifawaz1999 — CHANGES_REQUESTED** [review]
looks good thanks, just few comments. I dont mind making a diff PR for coverage to avoid confusion 

@MatthewMiddlehurst @TonyBagnall let me know what you think of my comment on the default saving

File: aeon/visualisation/results/_mcm.py (hunk 9 above)
Comment (aryanpola) [reply aeon/visualisation/results/_mcm.py]: Currently, all the formats will be saved when using default.

Comment (hadifawaz1999) [reply aeon/visualisation/results/_mcm.py]: i think by default it should not save

Comment (aryanpola) [reply aeon/visualisation/results/_mcm.py]: will update

**Comment by aryanpola** [comment]
I updated the default formats to None.

**Review by MatthewMiddlehurst — APPROVED** [review]
